govcd/ip_space.go

```diff
@@ -0,0 +1,227 @@
+}
+
+// GetAllIpSpaceSummaries retrieve summaries of all IP Spaces with an optional filter
+// Note. There is no API endpoint to get multiple IP Spaces with their full definitions. Only
+// "summaries" endpoint exists, but it does not include all fields. To retrieve complete structure
+// one can use `GetIpSpaceById` or `GetIpSpaceByName`
+func (vcdClient *VCDClient) GetAllIpSpaceSummaries(queryParameters url.Values) ([]*IpSpace, error) {
+	client := vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointIpSpaceSummaries
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	typeResponses := []*types.IpSpace{{}}
+	err = client.OpenApiGetAllItems(apiVersion, urlRef, queryParameters, &typeResponses, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wrap all typeResponses into IpSpace types with client
+	results := make([]*IpSpace, len(typeResponses))
+	for sliceIndex := range typeResponses {
+		results[sliceIndex] = &IpSpace{
+			IpSpace:   typeResponses[sliceIndex],
+			vcdClient: vcdClient,
+		}
+	}
+
+	return results, nil
+}
+
+// GetIpSpaceById retrieves IP Space with a given ID
+func (vcdClient *VCDClient) GetIpSpaceById(id string) (*IpSpace, error) {
+	if id == "" {
+		return nil, fmt.Errorf("IP Space lookup requires ID")
+	}
+
+	client := vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointIpSpaces
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint, id)
+	if err != nil {
+		return nil, err
+	}
+
+	response := &IpSpace{
+		vcdClient: vcdClient,
+		IpSpace:   &types.IpSpace{},
+	}
+
+	err = client.OpenApiGetItem(apiVersion, urlRef, nil, response.IpSpace, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+// GetIpSpaceByName retrieves IP Space with a given name
+// Note. It will return an error if multiple IP Spaces exist with the same name
+func (vcdClient *VCDClient) GetIpSpaceByName(name string) (*IpSpace, error) {
+	if name == "" {
+		return nil, fmt.Errorf("IP Space lookup requires name")
+	}
+
+	queryParameters := url.Values{}
+	queryParameters.Add("filter", "name=="+name)
+
+	filteredIpSpaces, err := vcdClient.GetAllIpSpaceSummaries(queryParameters)
+	if err != nil {
+		return nil, fmt.Errorf("error getting IP Spaces: %s", err)
+	}
+
+	singleIpSpace, err := oneOrError("name", name, filteredIpSpaces)
+	if err != nil {
+		return nil, fmt.Errorf("error ")
+	}
+
+	return vcdClient.GetIpSpaceById(singleIpSpace.IpSpace.ID)
+}
+
+// GetIpSpaceByNameAndOrgId retrieves IP Space with a given name in a particular Org
+// Note. Only PRIVATE IP spaces belong to Orgs
+func (vcdClient *VCDClient) GetIpSpaceByNameAndOrgId(name, orgId string) (*IpSpace, error) {
+	if name == "" {
```

Comment:
Just wondering whether we should also check `orgId == ""`

Reply:
Added